src/command.cpp

```diff
@@ -430,28 +430,44 @@ Command::CmdFunction * Command::find_command(const uint8_t device_type, const ui
 
     for (auto & cf : cmdfunctions_) {
         if (Helpers::toLower(cmd) == Helpers::toLower(cf.cmd_) && (cf.device_type_ == device_type) && (!device_id || cf.device_id_ == device_id)
-            && (!(flag & 0x3F) || (flag & 0x3F) == (cf.flags_ & 0x3F))) {
+            && (flag & 0x3F) == (cf.flags_ & 0x3F)) {
             return &cf;
         }
     }
 
     return nullptr; // command not found
 }
 
-void Command::erase_command(const uint8_t device_type, const char * cmd) {
+void Command::erase_command(const uint8_t device_type, const char * cmd, uint8_t flag) {
     if ((cmd == nullptr) || (strlen(cmd) == 0) || (cmdfunctions_.empty())) {
         return;
     }
     auto it = cmdfunctions_.begin();
     for (auto & cf : cmdfunctions_) {
-        if (Helpers::toLower(cmd) == Helpers::toLower(cf.cmd_) && (cf.device_type_ == device_type)) {
+        if (Helpers::toLower(cmd) == Helpers::toLower(cf.cmd_) && (cf.device_type_ == device_type) && ((flag & 0x3F) == (cf.flags_ & 0x3F))) {
             cmdfunctions_.erase(it);
             return;
         }
         it++;
     }
 }
 
+// get the tagged command
+std::string Command::tagged_cmd(std::string cmd, const uint8_t flag) {
```

Comment:
it's better to make this a `const std::string & cmd` so the reference is passed to the function and not the whole object class. I'll change it as I need to make some changes to get standalone compiling